lib/message-handler.js

```diff
@@ -360,6 +361,30 @@ class MessageHandler {
                                 return html;
                             })
                             .filter(html => html);
+
+                        messageData.intro = htmlToText(messageData.html.join(''))
+                            // assume we get the intro text from first 2 kB
```

Comment:
You are calling this set of regexes, etc, from two different places (both from the plaintext and HTML section), so it might make sense to create a function to convert text input into an intro string instead of defining the same regexes twice.